app/upgrades.go

```diff
@@ -4,41 +4,141 @@ import (
+func (app *WasmApp) NextUpgradeHandler(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (vm module.VersionMap, err error) {
+	sdkCtx := sdktypes.UnwrapSDKContext(ctx)
+	sdkCtx.Logger().Info("running module migrations", "name", plan.Name)
+
+	// Set the new parameters for mint and staking
+	if err := app.V19StakingForceMinimumCommission(ctx, plan); err != nil {
+		panic(fmt.Sprintf("failed set minimum commissions: %s", err))
+	}
+
+	// Set the new parameters for mint
+	if err := app.V19MintParamsChange(sdkCtx, plan); err != nil {
+		panic(fmt.Sprintf("failed to run mint module migrations: %s", err))
+	}
 
-			app.Logger().Info("setting upgrade store loaders")
-			app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
+	// Add query and swap epochs to the feeabs module
+	if err := app.V19FeeabsEpochAdd(sdkCtx, plan); err != nil {
+		panic(fmt.Sprintf("failed to run feeabs module epoch additions: %s", err))
+	}
+
+	// Run the migrations for all modules
+	migrations, err := app.ModuleManager.RunMigrations(ctx, app.Configurator(), fromVM)
+	if err != nil {
+		panic(fmt.Sprintf("failed to run migrations: %s", err))
+	}
+
+	return migrations, err
+}
+
+// V19StakingParamsChange is a migration function that sets the minimum commission rate for validators to 0.05
+func (app *WasmApp) V19StakingForceMinimumCommission(ctx context.Context, plan upgradetypes.Plan) (err error) {
+	// Get Staking params
+	stakingParams, err := app.StakingKeeper.GetParams(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get staking params %s", err)
+	}
+
+	// Set the minimum commission rate to 0.05
+	minCommission := math.LegacyMustNewDecFromStr("0.05")
+	stakingParams.MinCommissionRate = minCommission
+	err = app.StakingKeeper.SetParams(ctx, stakingParams)
+	if err != nil {
+		return fmt.Errorf("failed to set staking params %s", err)
+	}
+
+	// Iterate over all validators and update their commission rate if it's less than 0.05
+	err = app.StakingKeeper.IterateValidators(ctx, func(_ int64, validator stakingtypes.ValidatorI) (stop bool) {
+		if validator.GetCommission().LT(minCommission) {
+			val := validator.(stakingtypes.Validator)
+			_, err = app.StakingKeeper.UpdateValidatorCommission(ctx, val, minCommission)
+			if err != nil {
+				return true
+			}
 		}
+		return false
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to update validator commission %s", err)
+	}
+	return nil
+}
+
+// V19MintUpgrade is a migration function that sets the minting parameters for the xion mint module
+func (app *WasmApp) V19MintParamsChange(sdkCtx sdktypes.Context, plan upgradetypes.Plan) (err error) {
+	sdkCtx.Logger().Info("running mint module migrations", "name", plan.Name)
+
+	// Retrieve xion mint params
+	p, err := app.XionKeeper.XionLegacyMintKeeper.GetParams(sdkCtx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve params from legacy Xion Mint: %s", err)
 	}
+	// Prepare subspace for mint module migration
```

Comment:
This shouldn't be how you update the mint params.
It should be similarly done as for staking. i.e: GetParams(); change; SetParams().
The params set was only used during the migrations of x/mint params from x/params to their own keeper.

Reply:
The reasoning for this roundabout way to get the params has to do with the design for the xion mint keeper which uses a different store for its params, this way we can retrieve the legacy params store them in the mintSubspace so that the v53 mint module migration can run without any customization.

Otherwise we would need to make a mint module migration to store the params in the subspace and then another one to migrate to the SDK mint module.

Reply:
but have you checked it worked? The param set is never read anymore by the SDK mint module. It was read once here: https://github.com/cosmos/cosmos-sdk/blob/main/x/mint/migrations/v2/migrate.go#L16-L37, when going to v0.47 to v0.50 but that won't trigger again from v0.50 to v0.53.